src/Agent.Worker/TaskManager.cs

```diff
@@ -351,6 +351,13 @@ private void CheckForTaskDeprecation(IExecutionContext executionContext, Pipelin
                 }
 
                 executionContext.Warning(deprecationMessage);
+
+                var tailoredDeprecationMessage = taskJson["tailoredDeprecationMessage"];
```

Comment:
I think the task.json attribute can be just `deprecationMessage`. The fact that it is in task.json already implies it is tailored to the task.